src/index.js

```diff
@@ -272,29 +272,29 @@ export default class Undo {
       if (this.blockWasDeleted(state, nextState)) {
         this.insertDeletedBlock(state, nextState, index);
       } else if (this.blockWasSkipped(index, nextIndex, state, nextState)) {
-        this.blocks.delete(nextIndex);
+        await this.blocks.delete(nextIndex);
         this.caret.setToBlock(index, "end");
       } else if (blockCount > state.length) {
-        this.blocks
+        await this.blocks
```

Comment:
This `await` is not needed because the Promise is already being handled by `then` function.

Please review all cases where this applies.

Reply:
Sure!, actually the `then` was the not necessary code. When `then` is used, the rest of the code below the promise handle will be executed no matter the promise response status.